src/routers/handlers/yourCompanies/yourCompanies.ts

```diff
@@ -3,34 +3,50 @@ import { GenericHandler } from "../generic";
 import logger from "../../../lib/Logger";
 import {
     LANDING_URL,
+    MANAGE_AUTHORISED_PEOPLE_URL,
+    TRUE,
     YOUR_COMPANIES_ADD_COMPANY_URL,
     YOUR_COMPANIES_LANG
 } from "../../../constants";
 import { getTranslationsForView } from "../../../lib/utils/translations";
+import { Associations } from "../../../types/associations";
+import { getUserAssociations } from "../../../services/userCompanyAssociationService";
+import { getLoggedInUserEmail } from "../../../lib/utils/sessionUtils";
 
 export class YourCompaniesHandler extends GenericHandler {
 
-    execute (req: Request, response: Response): Promise<Object> {
+    async execute (req: Request, response: Response): Promise<Object> {
         logger.info(`GET request to serve Your Companies landing page`);
         // ...process request here and return data for the view
-        this.viewData = this.getViewData(this.userHasCompanies());
+        const userEmailAddress = getLoggedInUserEmail(req.session);
+        const userAssociations: Associations = await getUserAssociations(userEmailAddress);
+        this.viewData = this.getViewData(userAssociations);
         this.viewData.lang = getTranslationsForView(req.t, YOUR_COMPANIES_LANG);
         return Promise.resolve(this.viewData);
     }
 
-    private userHasCompanies (): boolean {
-        // TODO: The logic will be added as part of IDVA6-337
-        return false;
-    }
+    private getViewData (userAssociations: Associations): any {
+        const viewData: Record<string, any> = {
+            buttonHref: YOUR_COMPANIES_ADD_COMPANY_URL,
+            feedbackSource: LANDING_URL
+        };
+
+        if (userAssociations?.items?.length > 0) {
+            const associationData: { text: string}[][] = [];
+            for (let index = 0; index < userAssociations.items.length; index++) {
```

Comment:
It is possible to use short hand for looping over an array like this
for (const element of userAssociations.items) {
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of

Reply:
I know this syntax but in this case it's of no use because I need the index to use it in the body of the loop.

Reply:
Ok, I didn't realise you had to use the index, was thinking you could just push on to associationData.